doc/contribute/index.rst

```diff
@@ -362,6 +362,47 @@ it to contain:
     set -e exec
     exec git diff --cached | ${ZEPHYR_BASE}/scripts/checkpatch.pl -
 
+Instead of running Checkpatch at each commit, you may prefer to run it only
+before pushing on zephyr repo. To do this, make the file
+*$ZEPHYR_BASE/.git/hooks/pre-push* executable and edit it to contain:
+
+.. code-block:: bash
+
+    #!/bin/sh
+    set -e exec
+    remote="$1"
+    url="$2"
+
+    z40=0000000000000000000000000000000000000000
+
+    echo "Run push hook"
+
+    while read local_ref local_sha remote_ref remote_sha
+    do
+        if [ "$local_sha" = $z40 ]
+        then
+                # Handle delete
+                :
+        else
+                if [ "$remote_sha" = $z40 ]
+                then
+                        # New branch, examine all commits
+                        range="$local_sha"
+                else
+                        # Update to existing branch, examine new commits
+                        range="$remote_sha $local_sha"
+                fi
+
+                echo "Perform check patch"
+                exec git diff $range | /local/mcu/zephyr/zephyr-project/scripts/checkpatch.pl -
```

Comment:
Hmmm, I suspect this path ``/local/mcu/zephyr/zephyr-project/scripts/checkpatch.pl`` should be ``${ZEPHYR_BASE}/scripts/checkpatch.pl``.   Has this been tested for Windows too?

Reply:
You're suspecting correctly, txs